kafka/consume.go

```diff
@@ -79,39 +85,66 @@ func NewConsumer(log logrus.FieldLogger, conf Config, opts ...ConfigOpt) (*Confl
 		conf.Timeout = DefaultTimout
 	}
 
-	part := int32(-1)
-	if conf.Consumer.Partition != nil {
-		part = *conf.Consumer.Partition
-	}
-
 	cc := &ConfluentConsumer{
-		c:         consumer,
-		conf:      conf,
-		log:       log,
-		partition: part,
+		c:    consumer,
+		conf: conf,
+		log:  log,
 	}
 
-	if cc.partition < 0 {
+	if cc.conf.Consumer.Partition == nil {
```

Comment:
does this mean that I'd have configure on a partition if I wanted to partition by a key? 
```
conf := ConsumerConfig {
  Partition: 1, // anything will do
  PartitionKey: "what I actually care about",
  PartitionAlgorithm: "actually have to set this", 
}
```

So I think if you did a 
```
if cc.conf.Consumer.Partition == nil && cc.conf.Consumer.PartitionKey == "" {
```

Reply:
So we only need the partition key and the partition algorithm

Reply:
So if we default the alg then key || id is what we would need. But if you set only key here you'd `Subscribe` and not do the `Assign`